src/connector/builder.rs

```diff
@@ -169,6 +173,15 @@ impl ConnectorBuilder<WantsProtocols1> {
             enable_http1: false,
         })
     }
+
+    /// Override expected server name
```

Comment:
I think you should pick one stage where this can be set, instead of adding it to all three. Please rename the method to `with_server_name()` and explain in the documentation that this overrides the server name derived from the destination URL's host name.

Reply:
I left only one insance, at the `WantsProtocols1` stage